server/connectors/api-connector-jdbc/src/main/scala/com/prisma/api/connector/jdbc/impl/DeleteNodeInterpreters.scala

```diff
@@ -111,16 +111,18 @@ trait CascadingDeleteSharedStuff {
       parentIds: Vector[IdGCValue],
       visitedModels: Vector[Model]
   ): DBIO[Unit] = {
+
     for {
       ids            <- mutationBuilder.getNodeIdsByParentIds(parentField, parentIds)
+      groupedIds     = ids.grouped(10000).toVector
       model          = parentField.relatedModel_!
       _              = if (visitedModels.contains(model)) throw APIErrors.CascadingDeletePathLoops()
       nextCascadings = model.cascadingRelationFields.filter(_ != parentField)
-      childActions   = nextCascadings.map(field => recurse(mutationBuilder, field, ids, visitedModels :+ model))
+      childActions   = nextCascadings.map(field => DBIO.seq(groupedIds.map(recurse(mutationBuilder, field, _, visitedModels :+ model)): _*))
```

Comment:
This line is a bit hard to read. I guess a for comprehension could make it easier to understand. Can you try to refactor it?